internal/handlers/utils_test.go

```diff
@@ -0,0 +1,137 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/image"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/registry"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	registryPort = "5000/tcp"
+	authUser     = "user"
+	authPass     = "password"
+	registryURI  = "ghcr.io/rancher-sandbox/sbombastic/test-assets"
+	imageName    = "golang"
+	tag          = "1.12-alpine"
+	platform     = "linux/amd64"
+	digest       = "sha256:1782cafde43390b032f960c0fad3def745fac18994ced169003cb56e9a93c028"
+)
+
+type RegistryTestSuite struct {
+	registry    *registry.RegistryContainer
+	registryURL string
+	ctx         context.Context
+}
+
+func setupPrivateRegistry(t *testing.T) *RegistryTestSuite {
```

Comment:
there is no need to accept testing.T, we could just return error from here and let the caller deal with thee assertions/require

Reply:
Just accept a logger as a constructor parameter and store the slog logger inside the struct field